backend/app/Services/FileParser.php

```diff
@@ -2,31 +2,186 @@
 
   namespace App\Services;
 
+  use App\Episode;
   use App\Item;
+  use Illuminate\Database\Eloquent\Collection;
 
   class FileParser {
 
-    protected $response;
+    private $item;
+    private $episode;
+    private $tmdb;
+    private $storage;
 
-    public function __construct()
+    public function __construct(Item $item, Episode $episode, TMDB $tmdb, Storage $storage)
     {
-      $this->response = $this->parseMediaFiles();
-      $this->item = new Item();
+      $this->item = $item;
+      $this->episode = $episode;
+      $this->tmdb = $tmdb;
+      $this->storage = $storage;
     }
 
-    public function parseMediaFiles()
+    /**
+     * Make a request to flox-file-parser and get local files data.
+     *
+     * @return array
+     */
+    public function fetch()
     {
-      return file_get_contents(base_path('tests/fixtures/media_files.json'));
+      return json_decode(
+        file_get_contents(base_path('tests/fixtures/Files/all.json'))
+      );
     }
 
-    public function getTmdbId($title)
+    /**
+     * Loop over local files and see if it can find them in database,
+     * otherwise create a new item.
+     *
+     * @param $files
+     */
+    public function store($files)
     {
-      $item = $this->item->where('title', $title)->orWhere('original_title', $title)->first(['id', 'tmdb_id']);
+      foreach($files as $type => $items) {
+        foreach($items as $item) {
+          $title = $this->isTvShow($item) ? $item->tv_title : $item->name;
 
-      if( ! $item) {
-        return null;
+          // See if file is already in our database.
+          if($found = $this->foundInDatabase($title, 'title')) {
+            $this->handleStatus($item, $found->tmdb_id);
+
+            continue;
+          }
+
+          // Otherwise make a new TMDb request.
+          $this->searchTmdb($title, $item);
+        }
+      }
+    }
+
+    /**
+     * Make a new request to TMDb and check against the database.
+     * Otherwise create a new item.
+     *
+     * @param $title
+     * @param $item
+     * @return bool|mixed
+     */
+    private function searchTmdb($title, $item)
```

Comment:
searchTmdb name is confusing. It only searchs db in the first line and then does other stuff. Either rename it or put the other stuff in its own method.